qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -290,7 +304,32 @@ def _set_waveform_params(self, **params_dict: Dict) ->None:
         string += ",".join(
             ["{:7e}".format(params_dict[param]) for param in param_names])
         self.parent.write_raw(string)
+        
+    def _get_duty_cycle(self):
+        """
+        Simple function to extract the duty cycle after checking waveform
+        """
+        self.waveform()
+        wf = self.waveform.get_latest()
+
+        if wf == 'PULS' or wf == 'SQU':
```

Comment:
the following is more pythonic, readable, future-proof
```suggestion
        if wf in ['PULS', 'SQU']:
```

Reply:
completely agree, looks way better